app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,74 @@
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+
+  # Request the event monitor to stop running.
+  def stop_event_monitor
+    stop_running
+  end
+
+  # Start monitoring for events. This method blocks forever until
+  # #stop_event_monitor is called.
+  def monitor_events
+    _log.info "#{log_prefix} Monitoring for events"
+
+    start_running
+    event_monitor_running
+    while running?
+      heartbeat
+      events = event_producer.poll
+      heartbeat
+
+      _log.info "#{log_prefix} Received #{events.size} events" unless events.empty?
+      events.each do |event|
+        _log.debug "#{log_prefix} Event: #{event}"
+        @queue.enq event if event_requires_refresh?(event)
```

Comment:
Oh and regarding 'refresh', in this case I meant that the event requires refreshing data in the DB from the google-provider. Given what you've said above, I'm guessing this method is no longer needed